sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/InvalidCastToInterface.cs

```diff
@@ -132,4 +131,44 @@ public void TestMethod(object obj)
             return null;
         }
     }
+
+    interface IFoo { }
+    interface IBar { }
+
+    class Foo : IFoo { }
+    class Bar : IBar { }
+    class FooBar : IFoo, IBar { }
+    sealed class FinalBar : IBar { }
+
+    class Other
+    {
+        public void Method()
+        {
+            IFoo ifoo;
+            IBar ibar;
+            Foo foo;
+            Bar bar;
+            FooBar foobar;
+            FinalBar finalbar;
+            object o;
+
+            o = (IFoo)bar;  // Noncompliant
+            o = (IFoo)ibar;

```

Comment:
I would add a comment on the ones already caught by the compiler. When reading the cases this seems weird why we don't raise when it is obvious we should.